src/interceptors/ClientRequest/MockHttpSocket.ts

```diff
@@ -248,34 +248,53 @@ export class MockHttpSocket extends MockSocket {
     }
 
     if (response.body) {
-      const reader = response.body.getReader()
-
-      while (true) {
-        const { done, value } = await reader.read()
-
-        if (done) {
-          break
+      try {
+        const reader = response.body.getReader()
+
+        while (true) {
+          const { done, value } = await reader.read()
+
+          if (done) {
+            break
+          }
+
+          // Flush the headers upon the first chunk in the stream.
+          // This ensures the consumer will start receiving the response
+          // as it streams in (subsequent chunks are pushed).
+          if (httpHeaders.length > 0) {
+            flushHeaders(value)
+            continue
+          }
+
+          // Subsequent body chukns are push to the stream.
+          this.push(value)
         }
+      } catch (error) {
+        if (error instanceof Error) {
```

Comment:
This is not a blocker, but I think we need to return 500 for any error.
Current implementation hides the error from the user in case it is not extends `Error`

Reply:
Since anything can be thrown, how to do propose handling arbitrary data here? 

I'd say thrown errors likely indicate issues, and are handled as 500 responses as such. If a non-error is thrown, it's likely to be an explicit intention of the user. Perhaps actually throwing that would be the right thing to do?

Reply:
I completely agree! I revisited by previous point and found it wrong. I'm alining the uncaught exceptions handling in #555, and then will update this branch to migrate the implementation from `NodeClientRequest.ts`.